example/src/store/reducers/pid.ts

```diff
@@ -103,75 +104,94 @@ const pidSlice = createSlice({
         ...asyncStatusInitial,
         isLoading: true,
       };
     });
 
-    /* Dispatched when a prepare CiE L3 flow params async thunk rejected.
+    /* Dispatched when a prepare PID flow params async thunk is rejected.
      * Resets the flow params and sets the credential state to hasError while resetting isLoading and hasError
      * for the requested credential.
      */
-    builder.addCase(prepareCieL3FlowParamsThunk.rejected, (state, action) => {
-      state.pidCiel3FlowParams = initialState.pidCiel3FlowParams;
-      state.pidAsyncStatus.cieL3 = {
+    builder.addCase(preparePidFlowParamsThunk.rejected, (state, action) => {
+      state.pidFlowParams = initialState.pidFlowParams;
+      const authMethod = action.meta.arg.authMethod;
+      state.pidAsyncStatus[authMethod] = {
         ...asyncStatusInitial,
         hasError: { status: true, error: action.error },
       };
     });
 
-    /* Dispatched when a continue CiE L3 flow async thunk resolves.
-     * Resets the flow params and sets the obtained credential and sets its state to isDone while resetting isLoading and hasError
+    /* Dispatched when a continue PID flow async thunk resolves.
+     * Sets the obtained credential and sets its state to isDone while resetting isLoading and hasError
      * for the requested credential.
      */
-    builder.addCase(continueCieL3FlowThunk.fulfilled, (state, action) => {
-      state.pidCiel3FlowParams = initialState.pidCiel3FlowParams;
+    builder.addCase(continuePidFlowThunk.fulfilled, (state, action) => {
+      state.pidFlowParams = initialState.pidFlowParams;
+      const cieL3IsLoading = state.pidAsyncStatus.cieL3.isLoading;
+      const spidIsLoading = state.pidAsyncStatus.spid.isLoading;
+      const authMethod = cieL3IsLoading
+        ? "cieL3"
+        : spidIsLoading
+        ? "spid"
+        : "spid";
       state.pid = action.payload;
-      state.pidAsyncStatus.cieL3 = {
+      state.pidAsyncStatus[authMethod] = {
         ...asyncStatusInitial,
         isDone: true,
       };
     });
 
-    /* Dispatched when a continue CiE L3 flow async thunk is pending.
-     * Sets the credential state to isLoading while resetting isDone and hasError
+    /* Dispatched when a continue PID flow async thunk resolves.
+     * Sets the obtained credential and sets its state to isDone while resetting isLoading and hasError
      * for the requested credential.
      */
-    builder.addCase(continueCieL3FlowThunk.pending, (state) => {
-      // Redundant as already set by prepareCieL3FlowParams but we want to be explicit and set the loading state
-      state.pidAsyncStatus.cieL3 = {
+    builder.addCase(continuePidFlowThunk.pending, (state) => {
+      // Redundant as already set by preparePidFlowParams but we want to be explicit and set the loading state
+      const cieL3IsLoading = state.pidAsyncStatus.cieL3.isLoading;
+      const spidIsLoading = state.pidAsyncStatus.spid.isLoading;
+      const authMethod = cieL3IsLoading
```

Comment:
I set "spid" as the default value in case neither `cieL3IsLoading` nor `spidIsLoading` is true. This ensures that `authMethod` always has a valid value, even if neither of the authentication methods is currently loading.